lib/HLSL/HLModule.cpp

```diff
@@ -371,7 +371,7 @@ bool HLModule::IsEntryThatUsesSignatures(llvm::Function *F) {
   auto propIter = m_DxilFunctionPropsMap.find(F);
   if (propIter != m_DxilFunctionPropsMap.end()) {
     DxilFunctionProps &props = *(propIter->second);
-    return props.IsGraphics() || props.IsCS();
+    return props.IsGraphics() || props.IsCS() || props.IsNode();
```

Comment:
This matches the equivalent function in DXModule. However, I'm a little unsure whether this is right in all cases. The way this function is used is often to determine if the shader is a library or not. Was this added for the purpose of this location? https://github.com/microsoft/DirectXShaderCompiler/blob/769b504be4748931254036e3250c1daad5a1991a/lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp#L4231

It's the only one that looks related, but it actually looks like returning true for this function will skip `LegalizeDxilInputOutputs`. Is that right? 

I'm particularly concerned about this usage: 
https://github.com/microsoft/DirectXShaderCompiler/blob/769b504be4748931254036e3250c1daad5a1991a/lib/HLSL/HLMatrixLowerPass.cpp#L467

Reply:
Yes. It is for SROA to flatten the struct input parameter.
I'll move the check to SROA.

Reply:
I don't think the usage in `HLMatrixLowerPass.cpp` is a problem.  I think we could go back to adding node to the `IsEntryThatUsesSignatures` function instead of checking separately here.

For `HLMatrixLowerPass.cpp`, I think the point is to skip casting from matrix type to lowered type if it's in a signature that gets flattened, since that would be handled elsewhere.  I don't think there's any harm in including node here because there are no scenarios where we expect argument pointers to be used for matrix where they would not be part of the regular flattened/lowered parameters, because we would never be accessing record objects through a pointer derived from the argument pointer.